packages/ts/react-form/src/index.ts

```diff
@@ -259,3 +267,20 @@ export function useFormPart<M extends AbstractModel>(model: M): UseFormPartResul
     field,
   };
 }
+
+/**
+ * Hook to access an array model part of a form. It provides the same API as `useFormPart`,
+ * but adds an `items` property that allows to iterate over the items in form of an array of models.
+ *
+ * @param model - The array model to access
+ * @returns The array model part of the form
+ */
+export function useFormArrayPart<M extends ArrayModel>(
+  model: M,
+): M extends ArrayModel<infer M1> ? UseFormArrayPartResult<M, M1> : never {
```

Comment:
With the suggestion above, you don't need to use `extends` here:
```suggestion
): UseFormArrayPartResult<M> {
```